utils/src/token.rs

```diff
@@ -2,39 +2,40 @@ use std::fmt::{self, Debug};
 
 use tokio_util::sync::CancellationToken;
 
-#[derive(Clone)]
-pub(crate) struct Token {
+/// A token which can be used to signal a cancellation request
```

Comment:
We are wrapping `CancellationToken` mainly to make its name less confusing in the cases we don't use it for cancelling futures. I think we shouldn't use the word `cancel` in our docs. Check other places too

Reply:
maybe something along the lines "A token which can be used for signalling something exactly once"

Reply:
Vague, I like it :P